src/App.js

```diff
@@ -93,29 +91,14 @@ class App extends React.Component {
   };
 
   // Form inputs are stateful and handled by update functions
-  updateFirstName(event) {
-    this.setState({
-      firstname: event.target.value
-    });
-  }
-  updateLastName(event) {
-    this.setState({
-      lastname: event.target.value
-    });
-  }
-  updateCompany(event) {
-    this.setState({
-      company: event.target.value
-    });
-  }
-  updateReason(event) {
-    this.setState({
-      reason: event.target.value
-    });
-  }
-  updateHost(event) {
+  update(event) {
+    const property = event.target.name;
+
     this.setState({
-      host: event.target.value
+      //firstname: event.target.value
+      if ( userData.hasOwnProperty(property) ) {
```

Comment:
There's an even simpler version of this here:
https://stackoverflow.com/questions/21029999/react-js-identifying-different-inputs-with-one-onchange-handler